src/lib/matcher.rs

```diff
@@ -407,3 +408,252 @@ impl MatcherFactory {
+            invert_match,
+            reverse_complement: false, // does not impact FixedStringSetMatcher::bases_match()
+            color: false,
+        };
+
+        let matcher = FixedStringSetMatcher {
+            patterns: patterns,
+            opts: opts,
+        };
+
+        assert_eq!(matcher.bases_match(bases.as_bytes()), expected);
+    }
+
+    // ############################################################################################
+    // Tests RegexMatcher::match_bases()
+    // ####################################################################################
+    #[rstest]
+    #[case(false, ".GG", "AGGT", true)] // regex is found in bases, and invert_match is false
+    #[case(true, ".GG", "AGGT", false)] // regex is found in bases, and invert_matches is true
+    #[case(true, ".GG", "CCC", true)] // regex is not found in bases, and invert_match is true
+    #[case(false, ".GG", "CCC", false)] // regex is bot found in bases, and invert_match is false
+    fn test_regex_matcher_match_bases(
+        #[case] invert_match: bool,
+        #[case] pattern: String,
+        #[case] bases: String,
+        #[case] expected: bool,
+    ) {
+        let opts = MatcherOpts {
+            invert_match,
+            reverse_complement: false, // does not impact RegexMatcher::bases_match()
+            color: false,
+        };
+
+        let matcher = RegexMatcher::new(&pattern, opts);
+        assert_eq!(matcher.bases_match(bases.as_bytes()), expected);
+    }
+
+    // ############################################################################################
+    // Tests RegexSetMatcher::bases_match()
+    // ####################################################################################
+    #[rstest]
+    #[case(false, vec!["A.T", ".GG"], "AGTGG", true)] // all patterns are found in bases, and invert_match is false
+    #[case(true, vec!["A.T", ".GG"], "AGTGG", false)] // all patterns are found in bases, and invert_matches is true
+    #[case(false, vec!["A.T", ".TT"], "AGTGG", true)] // some patterns are found in bases, and invert_matches is false
+    #[case(true, vec!["A.T", ".TT"], "AGTGG", false)] // some patterns are found in bases, and invert_matches is true
+    #[case(false, vec!["C.C", ".TT"], "AGTGG", false)] // no patterns are found in bases, and invert_matches is false
+    #[case(true, vec!["C.C", ".TT"], "AGTGG", true)] // no patterns are found in bases, and invert_matches is true
+    fn test_regex_set_matcher_match_bases(
+        #[case] invert_match: bool,
+        #[case] patterns: Vec<&str>,
+        #[case] bases: String,
+        #[case] expected: bool,
+    ) {
+        let opts = MatcherOpts {
+            invert_match,
+            reverse_complement: false, // does not impact RegexSetMatcher::bases_match()
+            color: false,
+        };
+
+        let matcher = RegexSetMatcher::new(patterns, opts);
+        assert_eq!(matcher.bases_match(bases.as_bytes()), expected);
+    }
+}
```

Comment:
invert_match and reverse_complement are now tested with read_match()
```suggestion
}
```